plugins/modules/aws_glue_crawler.py

```diff
@@ -0,0 +1,409 @@
+
+    return changed
+
+
+def create_or_update_glue_crawler(connection, module, glue_crawler):
+    '''
+    Create or update an AWS Glue crawler
+    '''
+
+    changed = False
+    params = dict()
+    params['Name'] = module.params.get('name')
+    params['Role'] = module.params.get('role')
+    params['Targets'] = module.params.get('targets')
+    if module.params.get('database_name') is not None:
+        params['DatabaseName'] = module.params.get('database_name')
+    if module.params.get('description') is not None:
+        params['Description'] = module.params.get('description')
+    if module.params.get('recrawl_policy') is not None:
+        params['RecrawlPolicy'] = module.params.get('recrawl_policy')
+    if module.params.get('role') is not None:
+        params['Role'] = module.params.get('role')
+    if module.params.get('schema_change_policy') is not None:
+        params['SchemaChangePolicy'] = module.params.get('schema_change_policy')
+    if module.params.get('table_prefix') is not None:
+        params['TablePrefix'] = module.params.get('table_prefix')
+    if module.params.get('targets') is not None:
+        params['Targets'] = module.params.get('targets')
+
+    if glue_crawler:
+        if _compare_glue_crawler_params(params, glue_crawler):
+            module.fail_json(msg='%s vs %s' % (params, glue_crawler))
+            try:
+                if not module.check_mode:
+                    connection.update_crawler(aws_retry=True, **params)
+                changed = True
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e)
+    else:
+        try:
+            if not module.check_mode:
+                connection.create_crawler(aws_retry=True, **params)
+            changed = True
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e)
+
+    glue_crawler = _get_glue_crawler(connection, module, params['Name'])
+
+    changed |= ensure_tags(connection, module, glue_crawler)
+
+    module.exit_json(changed=changed, **camel_dict_to_snake_dict(glue_crawler or {}, ignore_list=['SchemaChangePolicy', 'RecrawlPolicy', 'Targets']))
+
+
+def delete_glue_crawler(connection, module, glue_crawler):
+    '''
+    Delete an AWS Glue crawler
+    '''
+    changed = False
+
+    if glue_crawler:
+        try:
+            if not module.check_mode:
+                connection.delete_crawler(aws_retry=True, Name=glue_crawler['Name'])
+            changed = True
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            module.fail_json_aws(e)
+
+    module.exit_json(changed=changed)
+
+
+def main():
+
+    argument_spec = (
+        dict(
+            database_name=dict(type='str'),
+            description=dict(type='str'),
+            name=dict(required=True, type='str'),
+            recrawl_policy=dict(type='dict'),
```

Comment:
I'm not a big fan of leaving this as just `type='dict'`.

It would be better to use `options=dict(...suboption_definition...)` here.  Reading just the parameter descriptions I wasn't quite sure what you meant.

The gotcha with suboptions is that you may need to use "amazon.aws.module_utils.scrub_none_parameters" if only some of the options should be specified

Reply:
Thanks, I replaced this with:

```
            recrawl_policy=dict(type='dict', options=dict(
                recrawl_behavior=dict(type='str')
            )),
            schema_change_policy=dict(type='dict', options=dict(
                delete_behavior=dict(type='str'),
                update_behavior=dict(type='str')
            )),
```